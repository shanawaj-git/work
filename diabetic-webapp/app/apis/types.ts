export type OTPGenerationResponse = {
  success: boolean;
};
export type PrescriptionUploadResponse = {
  id: number;
};

export type AuthUser = {
  id: number;
  username: string;
  email: string;
  blocked: false;
  firstName: string;
  middleName: string;
  lastName: string;
  mobileNumber: string;
  emiratesId: string;
  reminderFrequency: number;
  createdAt: string;
  updatedAt: string;
};

export type OTPValidationResponse = {
  jwt: string;
  user: AuthUser;
};

export type APIClientOptions = {
  onSuccessfulAuthentication: (response: OTPValidationResponse) => void;
  onUnauthorizedAccess: () => void;
  loginUrl: string;
};

export enum AuthDataKeys {
  Jwt = 'auth_jwt',
  User = 'auth_user',
}

export type OrdersInitializeResponse = {
  id: number;
  schedule: string;
  amount: number;
  currency: string;
  status: string;
  address: Address;
  prescription: Prescription;
};

export type Address = {
  id?: number;
  flatOrVillaNumber: number;
  buildingOrVilla: string;
  street: string;
  area?: string;
  directions?: string;
  latitude: number;
  longitude: number;
};

export type Prescription = {
  id: number;
  firstName: string;
  lastName: string;
  status: string;
};

export type OrdersInitializeRequest = {
  prescriptionNumber: string;
  address: Address;
  schedule: string;
};

export enum OrderStatus {
  Pending = 'Pending',
  InsuranceApproved = 'Insurance Approved',
  RequirePayment = 'Require Payment',
  PaymentReceived = 'Payment Received',
  DeliveryScheduled = 'Delivery Scheduled',
  Fulfilled = 'Fulfilled',
  Failed = 'Failed',
  Canceled = 'Canceled',
}

export enum PaymentSubType {
  Mastercard = 'Mastercard',
  Visa = 'Visa',
  ApplePay = 'Apple Pay',
  GooglePay = 'Google Pay',
  Cash = 'Cash',
}
export enum PaymentType {
  Card = 'Card',
  Wallet = 'Wallet',
  Cash = 'Cash',
  CashOnDelivery = 'Cash On Delivery',
  CardOnDelivery = 'Card On Delivery',
}

export enum PaymentStatus {
  Created = 'Created',
  Processing = 'Processing',
  PaymentMethodRequired = 'Payment Method Required',
  ActionRequired = 'ActionRequired',
  Canceled = 'Canceled',
  CaptureRequired = 'CaptureRequired',
  Captured = 'Captured',
  PartiallyCaptured = 'PartiallyCaptured',
  Succeeded = 'Succeeded',
  Failed = 'Failed',
  Unknown = 'Unknown',
}

export enum Currency {
  AED = 'AED',
}

export type Payment = {
  id: number;
  paymentType: PaymentType;
  paymentSubType: PaymentSubType;
  status: string;
  maskedCardNumber: string;
  providerMetadata: object;
  createdAt: string;
};

export type Order = {
  id: number;
  amount: number;
  currency: Currency;
  status: OrderStatus;
  payment: Payment;
};

export type Pharmacy = {
  pharmacyName: string;
  supportContactNumber: string;
  scheduleWindowDays: number;
  deliveryHours: object;
};
