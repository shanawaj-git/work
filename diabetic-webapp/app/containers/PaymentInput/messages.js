import { defineMessages } from 'react-intl';

const scope = 'orders.payment.input';

export default defineMessages({
  headerPrefix: {
    id: `${scope}.header.prefix`,
    defaultMessage: 'Payment',
  },

  headerSuffix: {
    id: `${scope}.header.suffix`,
    defaultMessage: 'details',
  },

  amountLabel: {
    id: `${scope}.amount.label`,
    defaultMessage: 'Total to pay',
  },
});
