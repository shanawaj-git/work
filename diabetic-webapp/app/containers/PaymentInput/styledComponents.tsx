import styled from 'styled-components';
import { colorSelector } from 'themes';
import { Typography } from '@albathanext/design-system';

export const MainContainer = styled.div`
  display: flex;
  background-color: ${colorSelector('p400')};
  flex: 1 1 auto;
  flex-direction: column;
  overflow-y: auto;
  padding: 24px;
  max-width: 600px;
  height: 100%;
  padding-top: 22px;
  margin: auto;
  gap: 8px;
`;
export const LoaderContainer = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  height: 100%;
`;

export const MpcLogo = styled.img`
  display: flex;
  width: 54px;
  height: 45px;
  align-self: flex-end;
`;

export const TitleContainer = styled.div`
  display: flex;
  gap: 8px;
  padding-bottom: 16px;
  align-items: baseline;
`;

export const TitlePrefix = styled(Typography)`
  &&.MuiTypography-root {
    font-weight: 400;
    font-size: 32px;
    line-height: 36px;
    font-style: normal;
  }
`;

export const TitleSuffix = styled(Typography)`
  &&.MuiTypography-root {
    font-weight: 600;
    font-size: 32px;
    line-height: 36px;
    font-style: normal;
  }
`;

export const AmountContainer = styled.div`
  display: flex;
  padding-bottom: 16px;
  align-items: baseline;
  justify-content: space-between;
`;

export const AmountLabel = styled(Typography)`
  &&.MuiTypography-root {
    font-style: normal;
    font-weight: 600;
    font-size: 15px;
    line-height: 22px;
  }
`;

export const AmountValue = styled(Typography)`
  display: flex;
  justify-content: center;
  align-items: center;
  padding: 4px 12px;
  background: #020a0a;
  border-radius: 8px;
  &&.MuiTypography-root {
    color: #ffffff;
    text-transform: uppercase;
    font-style: normal;
    font-weight: 600;
    font-size: 15px;
    line-height: 22px;
    text-align: center;
    letter-spacing: 0.3px;
  }
`;
