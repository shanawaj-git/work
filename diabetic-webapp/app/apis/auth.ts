import { getAPIClientInstance } from './client';
import { OTPGenerationResponse, OTPValidationResponse } from './types';

const Urls = {
  GENERATE_OTP: '/api/users-permissions/otp',
  VALIDATE_OTP: '/api/auth/local',
};

enum Errors {
  UnknownUser='auth.user.unknown',
  InvalidOTP='auth.otp.invalid',
  ExpiredOTP='auth.otp.expired',
  WaitToRegenerateOTP='auth.error.otp.wait',
  MaxFailedAttempt='auth.error.otp.limit'
}

export const publicEndpointConfig = {
  isPublicUrl: true,
}

const generateOTP = async (mobileNumber: string): Promise<OTPGenerationResponse> => {
  try {
    const response = await getAPIClientInstance().post(Urls.GENERATE_OTP, {
      mobileNumber,
    }, publicEndpointConfig);

    return { success: response.data.success };
  } catch (err) {
    if (err.response?.status === 404) {
      throw new Error(Errors.UnknownUser);
    } else if (err.response?.status === 403) {
      if (err.response.data.error.details.errorKey === Errors.WaitToRegenerateOTP)
        throw new Error(Errors.WaitToRegenerateOTP);
    } else {
      throw err;
    }
  }
};

const validateOTP = async (mobileNumber: string, otp: string): Promise<OTPValidationResponse> => {
  try {
    const { data: { jwt, user } } = await getAPIClientInstance().post(Urls.VALIDATE_OTP, {
      identifier: mobileNumber,
      password: otp,
    }, publicEndpointConfig);

    return { jwt, user: {...user} };
  } catch (err) {
    const { response: { status = undefined } = {}} =  err;
    if (status === 400) {
      throw new Error(Errors.InvalidOTP);
    } else if (status === 403) {
      if (err.response.data.error.details.errorKey === Errors.MaxFailedAttempt)
        throw new Error(Errors.MaxFailedAttempt);
      if (err.response.data.error.details.errorKey === Errors.ExpiredOTP)
        throw new Error(Errors.ExpiredOTP);
    } else {
      throw err;
    }
  }
};

export {
  generateOTP,
  validateOTP,
  Urls as AuthApiUrls,
  Errors as AuthApiErrors,
};
