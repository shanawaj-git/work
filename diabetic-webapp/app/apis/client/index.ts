import { APIClientOptions } from 'apis/types';
import axios from 'axios';
import { addInterceptors } from './addInterceptors';
import * as environmentConfig from '../../configs/environmentConfig';

let httpClient;
// eslint-disable-next-line no-unused-vars
const configureAPIClient = (options: APIClientOptions) => {
  // TODO: intercept response and add callback support for login, auth failed
  httpClient = axios.create({
    baseURL:
      environmentConfig.getEnvironmentConfig().API_BASE_URL ||
      'http://localhost:1337',
    timeout: 30000, // 30 seconds
    // TODO: pass Authorization once login is implemented
  });
  addInterceptors(httpClient, options);
  return httpClient;
};

const getAPIClientInstance = () => {
  if (!httpClient) {
    throw new Error('API Client is not intialized');
  }
  return httpClient;
};

export { configureAPIClient, getAPIClientInstance, APIClientOptions };
