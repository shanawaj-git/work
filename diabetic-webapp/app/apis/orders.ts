import { getAPIClientInstance } from './client';
import {
  OrdersInitializeResponse,
  OrdersInitializeRequest,
  Order,
  PaymentType,
} from './types';

const Urls = {
  ORDERS_INITIALIZE: '/api/orders/initialize',
  orderDetails: (id: string) => `/api/orders/${id}`,
  confirmPayment: (id: string) => `/api/orders/${id}/payments/confirmation`,
};

enum Errors {
  PrescriptionNotFound = 'order.prescription.not.found',
  OrderAlreadyExists = 'order.already.exits',
}

const initializeOrder = async (
  orderInitializeRequest: OrdersInitializeRequest,
): Promise<OrdersInitializeResponse> => {
  try {
    return (await getAPIClientInstance().post(
      Urls.ORDERS_INITIALIZE,
      orderInitializeRequest,
    )).data;
  } catch (error) {
    const { response: { status = undefined } = {} } = error;
    if (status === 404) {
      throw new Error(Errors.PrescriptionNotFound);
    } else if (status === 409) {
      throw new Error(Errors.OrderAlreadyExists);
    } else {
      throw error;
    }
  }
};

const fetchOrderDetails = async (orderId: string): Promise<Order> => {
  return (await getAPIClientInstance().get(Urls.orderDetails(orderId))).data;
};

const confirmPayment = async (
  orderId: string,
  type?: PaymentType,
): Promise<Order> => {
  const args: [any] = [Urls.confirmPayment(orderId)];
  if(type) {
    args.push({
      paymentMethod: type,
    });
  }
  return (await getAPIClientInstance().put(...args)).data;
};

export {
  initializeOrder,
  fetchOrderDetails,
  confirmPayment,
  Urls as OrdersApiUrls,
  Errors as OrderApiErrors,
};
