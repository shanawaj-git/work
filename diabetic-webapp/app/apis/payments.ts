import { getAPIClientInstance } from './client';
import { Payment } from './types';

export const Urls = {
  PAYMENTS_INITIALIZE: '/api/payments/initialize',
};

export const initializePayment = async (orderId: string): Promise<Payment> =>
  (await getAPIClientInstance().post(Urls.PAYMENTS_INITIALIZE, {
    orderId,
  })).data;
