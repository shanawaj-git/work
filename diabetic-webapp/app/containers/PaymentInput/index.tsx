import React, { useEffect, useState } from 'react';
import { injectIntl, intlShape } from 'react-intl';
import PropTypes, { InferProps } from 'prop-types';
import { useParams } from 'react-router-dom';

import Loader from 'components/moclecules/Loader';
import PaymentForm from 'components/organism/PaymentForm';
import * as ordersApi from '../../apis/orders';
import * as paymentsApi from '../../apis/payments';

import { Order, OrderStatus, PaymentStatus } from '../../apis/types';
import {
  TitleContainer,
  LoaderContainer,
  MainContainer,
  MpcLogo,
  TitlePrefix,
  TitleSuffix,
  AmountContainer,
  AmountLabel,
  AmountValue,
} from './styledComponents';
import messages from './messages';
const mpcGrayLogo = require('images/mpc_gray.svg');

const PAYMENT_STATUS_PAGE = 'status';

const validOrderStatuses = [
  OrderStatus.InsuranceApproved,
  OrderStatus.RequirePayment,
];

export const SEARCH_PARAM_CONFIRMATION = 'confirmation';

const ProgressBar = () => (
  <LoaderContainer>
    <Loader size={90} />
  </LoaderContainer>
);

const Form = ({ order, intl }: { order: Order; intl: any }) => {
  // eslint-disable-next-line dot-notation
  const clientSecret = order.payment.providerMetadata['client_secret'];
  const RETURN_URL = `${window.location.href}?${SEARCH_PARAM_CONFIRMATION}`;
  return (
    <>
      <TitleContainer>
        <TitlePrefix>{intl.formatMessage(messages.headerPrefix)}</TitlePrefix>
        <TitleSuffix>{intl.formatMessage(messages.headerSuffix)}</TitleSuffix>
      </TitleContainer>
      <AmountContainer>
        <AmountLabel>{intl.formatMessage(messages.amountLabel)}</AmountLabel>
        <AmountValue>{`${order.currency} ${order.amount}`}</AmountValue>
      </AmountContainer>
      <PaymentForm clientSecret={clientSecret} returnUrl={RETURN_URL} />
    </>
  );
};

const PaymentInput = ({ history, intl, location }: PaymentInputProps) => {
  const { orderId } = useParams();

  const [orderDetails, setOrderDetails] = useState<Order>();

  const redirectToOrderStatus = () => history.push(PAYMENT_STATUS_PAGE);

  const initializePaymentDetails = async () =>
    paymentsApi.initializePayment(orderId);

  const fetchOrderDetails = async (id: string) => {
    try {
      const order = await ordersApi.fetchOrderDetails(id);
      if (!validOrderStatuses.includes(order.status)) {
        redirectToOrderStatus();
        return;
      }
      if (!order.payment || !order.payment.providerMetadata) {
        const payment = await initializePaymentDetails();
        order.payment = { ...payment };
      }

      if (order.payment.status === PaymentStatus.PaymentMethodRequired) {
        setOrderDetails(order);
      } else {
        redirectToOrderStatus();
      }
    } catch (err) {
      // eslint-disable-next-line prettier/prettier
      const responseStatus = err?.response?.status;
      if(responseStatus !== 401 && responseStatus !== 403)
        redirectToOrderStatus();
    }
  };

  const confirmPayment = async () => {
    try {
      await ordersApi.confirmPayment(orderId);
    } catch (err) {
      // In case of error too it should redirect to the status page.
    } finally {
      redirectToOrderStatus();
    }
  };

  const isConfirmation = new URLSearchParams(location.search).has(
    SEARCH_PARAM_CONFIRMATION,
  );

  useEffect(() => {
    if (isConfirmation) {
      confirmPayment();
    } else {
      fetchOrderDetails(orderId);
    }
  }, [orderId]);

  const initialized = !!orderDetails;

  return (
    <MainContainer>
      <MpcLogo src={mpcGrayLogo} />
      {!initialized && <ProgressBar />}
      {initialized && <Form order={orderDetails} intl={intl} />}
    </MainContainer>
  );
};

const PaymentInputPropTypes = {
  history: PropTypes.shape({
    push: PropTypes.func.isRequired,
  }),
  location: PropTypes.shape({
    search: PropTypes.string.isRequired,
  }),
  intl: intlShape,
};

type PaymentInputProps = InferProps<typeof PaymentInputPropTypes>;

PaymentInput.propTypes = PaymentInputPropTypes;

export default injectIntl(PaymentInput);
