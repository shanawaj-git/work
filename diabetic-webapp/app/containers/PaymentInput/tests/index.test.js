import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import { IntlProvider } from 'react-intl';
import { render, wait } from 'react-testing-library';

import mockStripe from 'tests/utils/mockStripe';
import { OrderStatus, PaymentStatus } from 'apis/types';
import * as ordersApi from 'apis/orders';
import * as paymentsApi from 'apis/payments';

// eslint-disable-next-line import/first
import PaymentInput, { SEARCH_PARAM_CONFIRMATION } from '../index';

jest.mock('@stripe/react-stripe-js', () => mockStripe());

const mockOrderId = '1000';

const mockPayment = {
  id: 1234,
  paymentType: 'Visa',
  maskedCardNumber: 'xxxxxxxxxxxx1234',
  status: PaymentStatus.PaymentMethodRequired,
  providerMetadata: {
    client_secret: 'some dummy secret',
  },
  createdAt: '2022-10-20T11:17:43.561Z',
};

const mockOrder = {
  id: mockOrderId,
  amount: 123.33,
  currency: 'AED',
  status: OrderStatus.RequirePayment,
  payment: mockPayment,
};

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'), // use actual for all non-hook parts
  useParams: () => ({
    orderId: mockOrderId,
  }),
}));

describe('<PaymentInput />', () => {
  // eslint-disable-next-line no-global-assign
  window = Object.assign(window, {
    visualViewport: {
      addEventListener: jest.fn(),
      removeEventListener: jest.fn(),
    },
  });

  // eslint-disable-next-line no-unused-vars
  let mockHistory;
  let fetchOrderDetailsSpy;
  let initializePaymentSpy;
  let confirmPaymentSpy;

  beforeEach(() => {
    mockHistory = {
      push: jest.fn(),
    };
    fetchOrderDetailsSpy = jest
      .spyOn(ordersApi, 'fetchOrderDetails')
      .mockResolvedValue(mockOrder);

    initializePaymentSpy = jest
      .spyOn(paymentsApi, 'initializePayment')
      .mockResolvedValue(mockPayment);
    confirmPaymentSpy = jest
      .spyOn(ordersApi, 'confirmPayment')
      .mockResolvedValue(mockPayment);
  });

  afterEach(() => {
    mockHistory = null;
    fetchOrderDetailsSpy.mockRestore();
    initializePaymentSpy.mockRestore();
    confirmPaymentSpy.mockRestore();
  });

  // eslint-disable-next-line react/prop-types
  const MockPaymentInput = ({ locale, search = '' }) => (
    <BrowserRouter>
      <IntlProvider locale={locale}>
        <PaymentInput history={mockHistory} location={{ search }} />
      </IntlProvider>
    </BrowserRouter>
  );

  it('should render and match the snapshot', async () => {
    const {
      container: { firstChild },
    } = render(<MockPaymentInput locale="en" />);
    await wait(() => {
      expect(firstChild).toMatchSnapshot();
    });
  });

  it('should render the loader while api calls are in progress', async () => {
    const {
      container: { firstChild },
    } = render(<MockPaymentInput locale="en" />);
    expect(firstChild).toMatchSnapshot();
  });

  it('should invoke fetchOrderDetails with the order id', async () => {
    render(<MockPaymentInput locale="en" />);
    await wait(() => {
      expect(fetchOrderDetailsSpy).toBeCalledWith(mockOrderId);
    });
  });

  it('should redirect to the payment status page if the order status is not valid', async () => {
    fetchOrderDetailsSpy.mockResolvedValue({
      ...mockOrder,
      status: OrderStatus.Pending,
    });

    render(<MockPaymentInput locale="en" />);
    await wait(() => {
      expect(mockHistory.push).toBeCalledWith('status');
    });
  });

  it('should redirect to the payment status page if the payment details status is not valid', async () => {
    fetchOrderDetailsSpy.mockResolvedValue({
      ...mockOrder,
      payment: {
        ...mockOrder.payment,
        status: PaymentStatus.Processing,
      },
    });

    render(<MockPaymentInput locale="en" />);
    await wait(() => {
      expect(mockHistory.push).toBeCalledWith('status');
    });
  });

  it('should invoke initializePayment api with the orderId if payment is not initialized', async () => {
    fetchOrderDetailsSpy.mockResolvedValue({
      ...mockOrder,
      payment: null,
    });

    render(<MockPaymentInput locale="en" />);
    await wait(async () => {
      await wait(() => {
        expect(initializePaymentSpy).toBeCalledWith(mockOrderId);
      });
    });
  });

  it('should invoke initializePayment api with the orderId if providerMetadata is not initialized', async () => {
    fetchOrderDetailsSpy.mockResolvedValue({
      ...mockOrder,
      payment: {
        ...mockPayment,
        providerMetadata: null,
      },
    });

    render(<MockPaymentInput locale="en" />);
    await wait(async () => {
      await wait(() => {
        expect(initializePaymentSpy).toBeCalledWith(mockOrderId);
      });
    });
  });

  it('should invoke confirmPayment api with the orderId and redirect to status page upon confirmation', async () => {
    render(
      <MockPaymentInput locale="en" search={`?${SEARCH_PARAM_CONFIRMATION}`} />,
    );
    await wait(() => {
      expect(confirmPaymentSpy).toBeCalledWith(mockOrderId);
      expect(mockHistory.push).toBeCalledWith('status');
    });
  });

  it('should show appropriate error when fetchOrderDetails is failed', async () => {
    fetchOrderDetailsSpy.mockRejectedValue(new Error('Some error.'));

    render(<MockPaymentInput locale="en" />);
    await wait(() => {
      expect(mockHistory.push).toBeCalledWith('status');
    });
  });

  it('should show appropriate error when initializePayment is failed', async () => {
    fetchOrderDetailsSpy.mockResolvedValue({
      ...mockOrder,
      payment: null,
    });
    initializePaymentSpy.mockRejectedValue(new Error('Some error'));

    render(<MockPaymentInput locale="en" />);
    await wait(() => {
      expect(mockHistory.push).toBeCalledWith('status');
    });
  });

  it('should show appropriate error when confirmPayment is failed', async () => {
    confirmPaymentSpy.mockRejectedValue(new Error('Some error'));

    render(
      <MockPaymentInput locale="en" search={`?${SEARCH_PARAM_CONFIRMATION}`} />,
    );
    await wait(() => {
      expect(mockHistory.push).toBeCalledWith('status');
    });
  });

  it('should not redirect to status page on Unauthorized error on fetching order details', async () => {
    fetchOrderDetailsSpy.mockRejectedValue({ response: { status: 401 } });

    render(<MockPaymentInput locale="en" />);
    await wait(() => {
      expect(mockHistory.push).not.toBeCalled();
    });
  });

  it('should not redirect to status page on Forbidden error on fetching order details', async () => {
    fetchOrderDetailsSpy.mockRejectedValue({ response: { status: 403 } });

    render(<MockPaymentInput locale="en" />);
    await wait(() => {
      expect(mockHistory.push).not.toBeCalled();
    });
  });
});
