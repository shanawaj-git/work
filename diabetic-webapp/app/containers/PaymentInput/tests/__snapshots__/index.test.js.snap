// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PaymentInput /> should render and match the snapshot 1`] = `
<div
  class="sc-htpNat gHbikN"
>
  <img
    class="sc-ifAKCX PtYSZ"
    src="IMAGE_MOCK"
  />
  <div
    class="sc-EHOje SMKFC"
  >
    <p
      class="MuiTypography-root MuiTypography-body1 sc-bZQynM gWMWXB css-1reifv3-MuiTypography-root"
    >
      Payment
    </p>
    <p
      class="MuiTypography-root MuiTypography-body1 sc-gzVnrw gIrCGn css-1reifv3-MuiTypography-root"
    >
      details
    </p>
  </div>
  <div
    class="sc-htoDjs jqrcbN"
  >
    <p
      class="MuiTypography-root MuiTypography-body1 sc-dnqmqq ewloC css-1reifv3-MuiTypography-root"
    >
      Total to pay
    </p>
    <p
      class="MuiTypography-root MuiTypography-body1 sc-iwsKbI guEjle css-1reifv3-MuiTypography-root"
    >
      AED 123.33
    </p>
  </div>
  <form
    class="sc-bdVaJa hskcns"
    id="payment-form"
  >
    <div />
    <div
      class="sc-bwzfXH dwOEUG"
    >
      <button
        class="root StyledButton-sc-1apep4b-0 hPQGaD MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-sghohy-MuiButtonBase-root-MuiButton-root"
        data-testid="continueButton"
        id="submit"
        tabindex="0"
        type="submit"
      >
        <div
          style="visibility: visible;"
        >
          Continue
        </div>
      </button>
    </div>
  </form>
</div>
`;

exports[`<PaymentInput /> should render the loader while api calls are in progress 1`] = `
<div
  class="sc-htpNat gHbikN"
>
  <img
    class="sc-ifAKCX PtYSZ"
    src="IMAGE_MOCK"
  />
  <div
    class="sc-bxivhb ihiqzF"
  >
    <div
      class="MuiBox-root css-15t75f3"
    >
      <span
        aria-valuenow="100"
        class="MuiCircularProgress-root MuiCircularProgress-determinate MuiCircularProgress-colorPrimary css-1e5kc5y-MuiCircularProgress-root"
        role="progressbar"
        style="width: 90px; height: 90px; transform: rotate(-90deg);"
      >
        <svg
          class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
          viewBox="22 22 44 44"
        >
          <circle
            class="MuiCircularProgress-circle MuiCircularProgress-circleDeterminate css-oxts8u-MuiCircularProgress-circle"
            cx="44"
            cy="44"
            fill="none"
            r="21"
            stroke-width="2"
            style="stroke-dasharray: 131.947; stroke-dashoffset: 0.000px;"
          />
        </svg>
      </span>
      <svg
        style="position: absolute;"
      >
        <defs>
          <lineargradient
            id="my_gradient"
          >
            <stop
              offset="0%"
              stop-color="#00664f"
            />
            <stop
              offset="100%"
              stop-color="#c5c5c5"
            />
          </lineargradient>
        </defs>
      </svg>
      <span
        class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-1utiupp-MuiCircularProgress-root"
        role="progressbar"
        style="width: 90px; height: 90px;"
      >
        <svg
          class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
          viewBox="22 22 44 44"
        >
          <circle
            class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate MuiCircularProgress-circleDisableShrink css-79nvmn-MuiCircularProgress-circle"
            cx="44"
            cy="44"
            fill="none"
            r="21"
            stroke-width="2"
          />
        </svg>
      </span>
    </div>
  </div>
</div>
`;
